src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -21,51 +105,1972 @@ const ActorInit Boss_Fd_InitVars = {
+                part->color.g = sp8C[cInd].g;
+                part->color.b = sp8C[cInd].b;
+                part->opacity -= 0x14;
+                if (part->opacity <= 0) {
+                    part->opacity = 0;
+                    part->type = 0;
+                }
+            } else if ((part->type == 2) || (part->type == 5)) {
+                part->unk_34 += 0.55f;
+                part->unk_38 += 0.1f;
+                if (part->pos.y <= 100.0f) {
+                    part->type = 0;
+                }
+            } else if (3 == part->type) {
+                if (part->timer2 >= 8) {
+                    part->timer2 = 8;
+                    part->type = 0;
+                } else if (((part->timer1 & 1) != 0) || (Math_Rand_ZeroOne() < 0.3f)) {
+                    part->timer2++;
+                }
+            } else if (part->type == 4) {
+                diff.x = player->actor.posRot.pos.x - part->pos.x;
+                diff.y = (player->actor.posRot.pos.y + 30.0f) - part->pos.y;
+                diff.z = player->actor.posRot.pos.z - part->pos.z;
+                if ((this->animationTimers[3] == 0) && (sqrtf(SQ(diff.x) + SQ(diff.y) + SQ(diff.z)) < 20.0f)) {
+                    this->animationTimers[3] = 0x32;
+                    func_8002F6D4(globalCtx, NULL, 5.0f, part->kbAngle, 0.0f, 0x30);
+                    if (player->unk_A60 == 0) {
+                        for (i2 = 0; i2 < 18; i2++) {
+                            player->unk_A61[i2] = Math_Rand_S16Offset(0, 200);
+                        }
+                        player->unk_A60 = 1;
+                    }
+                }
+                if (part->timer2 == 0) {
+                    if (part->scale < 2.5f) {
+                        part->scale += part->unk_34;
+                        part->unk_34 += 0.08f;
+                    }
+                    if ((part->pos.y <= (part->unk_38 + 130.0f)) || (part->timer1 >= 10)) {
+                        part->accel.y = 5.0f;
+                        part->timer2 += 1;
+                        part->vel.y = 0.0f;
+                        part->accel.x = (part->vel.x * -25.0f) / 100.0f;
+                        part->accel.z = (part->vel.z * -25.0f) / 100.0f;
+                    }
+                } else {
+                    if (part->scale < 2.5f) {
+                        Math_SmoothScaleMaxF(&part->scale, 2.5f, 0.5f, 0.5f);
+                    }
+                    part->timer2++;
+                    if (part->timer2 >= 9) {
+                        part->type = 0;
+                    }
+                }
+            }
+        }
+    }
+}
+
+UNK_TYPE D_808D1A9C[] = { 0x04051DB0, 0x04051DB0, 0x040521B0, 0x040525B0, 0x040529B0,
+                          0x04052DB0, 0x040531B0, 0x040535B0, 0x040539B0 };
+
+void func_808CF898(BossFdParticle* ptr, GlobalContext* globalCtx) {
+    u8 flag = 0;
+    s16 i;
+    f32 pad;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    BossFdParticle* particle;
```

Comment:
Is `particle` required to match?
Since `ptr` seems like it's the most used maybe name them `particle` and `particle_`, just because `ptr` is a bit too generic?

Reply:
Yeah it is. It's weird that it works that way, but it really wants the variable that came from the function to be the iterator.

Reply:
What about renaming `ptr` then? I feel like it would make the code more readable

Reply:
renamed ptr to particle and particle to firstParticle.